### PR TITLE
docs(discovery): add DISCOVERY_PLUGINS.md

### LINKED ISSUE #939: [Story] Pluggable Discovery API
Parent: https://github.com/cryostatio/cryostat/issues/936

Internally we currently have the `MergingPlatformClient` that serves this goal, but only for the hardcoded and included implementations within Cryostat. We need to generify this and expose it as an authenticated API so that API clients can perform their own platform-specific discovery of environment and target nodes and then publish that information to Cryostat, where Cryostat will then store it in its internal database. This API should allow for Create, Read, Update, Delete semantics on the node resources. This will serve parent 1 and 5, and should address portions of 2 as well.

- [x] Implement Pluggable Discovery API (draft: #1026)
- [x] #1049
- [x] #1058
- [x] #1061

### PR DESCRIPTION
**Pull request by andrewazores**
Related to #1049
Closes #939

### CODE REVIEW

File: docs/DISCOVERY_PLUGINS.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# Discovery Plugins
+
+As of Cryostat v2.2, the `/api/v2.2/discovery` and related API endpoints can be
+used to register new external plugins which provide Cryostat with information
+about discoverable targets known to the plugin. For more details on the API
+request specifics, see HTTP_API.md. This document serves as a high-level
+description and explains the motiviations for this API.
```
Comment (tthvo) [inline docs/DISCOVERY_PLUGINS.md:7]: Typo at `motiviations`?

File: docs/DISCOVERY_PLUGINS.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+may modify their own applications to self-publish themselves to the discovery
+interface directly. This can be used as a supplement to or a replacement for
+Cryostat's built-in discovery mechanisms.
+
+## Client-side API
+
+On the client (Plugin) side, the discovery API requirements are a relatively low
+barrier. The client must be able to send HTTP requests to the Cryostat server,
+must be able to pass an `Authorization` header with acceptable Cryostat
+credentials, must be able to accept JSON formatted responses, and must be able
```
Comment (tthvo) [inline docs/DISCOVERY_PLUGINS.md:32]: Perhaps remove subsequent `must be able to`?

**Review by tthvo — APPROVED** [review]
Looks great to me :D

**Review by maxcao13 — APPROVED** [review]
Looks good to me as well.

**Pull request merged**